juju/juju
# Proxy support4register controller

juju `add-user` and `register` supports k8s proxy now.

## Checklist


## QA steps

```console
$ juju add-user admin1
User "admin1" added
Please send this command to admin1:
    juju register MIIEsxMGYWRtaW4xMBUTEzEwLjE1Mi4xODMuNzc6MTcwNzAEIL3EM-Zld6ce7FT9NyWwTs9HT2kUnFtzSaBQfI4smSjWEwJrMQyCBGp0eXBlOiBrdWJlcm5ldGVzLXBvcnQtZm9yd2FyZApjb25maWc6CiAgYXBpLWhvc3Q6IGh0dHBzOi8vMTI3LjAuMC4xOjE2NDQzCiAgY2EtY2VydDogIiIKICBuYW1lc3BhY2U6IGNvbnRyb2xsZXItazEKICByZW1vdGUtcG9ydDogIjE3MDcwIgogIHNlcnZpY2U6IGNvbnRyb2xsZXItc2VydmljZQogIHNlcnZpY2UtYWNjb3VudC10b2tlbjogZXlKaGJHY2lPaUpTVXpJMU5pSXNJbXRwWkNJNklteE1lV1ZJVFd3Mk56WjNTbmhUTTNwUlVuWmFRMUl3VDJOM1JFaEhUMkZtTmtreFJHZFhXbVV6UjNjaWZRLmV5SnBjM01pT2lKcmRXSmxjbTVsZEdWekwzTmxjblpwWTJWaFkyTnZkVzUwSWl3aWEzVmlaWEp1WlhSbGN5NXBieTl6WlhKMmFXTmxZV05qYjNWdWRDOXVZVzFsYzNCaFkyVWlPaUpqYjI1MGNtOXNiR1Z5TFdzeElpd2lhM1ZpWlhKdVpYUmxjeTVwYnk5elpYSjJhV05sWVdOamIzVnVkQzl6WldOeVpYUXVibUZ0WlNJNkltTnZiblJ5YjJ4c1pYSXRjSEp2ZUhraUxDSnJkV0psY201bGRHVnpMbWx2TDNObGNuWnBZMlZoWTJOdmRXNTBMM05sY25acFkyVXRZV05qYjNWdWRDNXVZVzFsSWpvaVkyOXVkSEp2Ykd4bGNpMXdjbTk0ZVNJc0ltdDFZbVZ5Ym1WMFpYTXVhVzh2YzJWeWRtbGpaV0ZqWTI5MWJuUXZjMlZ5ZG1salpTMWhZMk52ZFc1MExuVnBaQ0k2SWpObE5HWTNZek5qTFRsaVpUVXROR1V4TnkxaVl6RXpMV001TnpaaFpESXpZV1l3WVNJc0luTjFZaUk2SW5ONWMzUmxiVHB6WlhKMmFXTmxZV05qYjNWdWREcGpiMjUwY205c2JHVnlMV3N4T21OdmJuUnliMnhzWlhJdGNISnZlSGtpZlEuUTFoQThNUTh3TDQxcEtUYlZwYk43dFZZN0pyX2UzQ0pYa1Q4aVZzdERtTG1DMTE4cndlZzNuQVlJX282ZmdJb1lOODFYcDFWdUlFUjdBWEo2RGQxMmo2WFhuWEw0R3ZVRkQ3bjdwWWtQdE9ZZmhKSmpwdENINFJaWW1HQ1FobUdsVGJTZzdkUF9fVDFBN194NWNKbm83V0RHM3pDSUlJMllIdVl5ZkxrY2lHaXlzMWh2QllOQS04Z3h5anZ4RU12NUV6OFNNaEJkQ3dOaVY1em5nVjJVblA4R1F3WExHYjBkUzAyZlFzZTBCZHpfWHFBcjhSa3d4dlh1a2FtX3dHRWoyX2tBVllaN3VEbUVzLXA1VU15RjBXZDJObEFyUzJyWDVaT3czVU5pak5HN1p0VV9hUl81UF8yWkpXRG05MEJVS2dEalRvQ3FUZlNqYzdMeHpjY1BnCgAA

"admin1" has not been granted access to any models. You can use "juju grant" to grant access.

$ export JUJU_DATA=/tmp/test-juju

$ juju register MIIEsxMGYWRtaW4xMBUTEzEwLjE1Mi4xODMuNzc6MTcwNzAEIL3EM-Zld6ce7FT9NyWwTs9HT2kUnFtzSaBQfI4smSjWEwJrMQyCBGp0eXBlOiBrdWJlcm5ldGVzLXBvcnQtZm9yd2FyZApjb25maWc6CiAgYXBpLWhvc3Q6IGh0dHBzOi8vMTI3LjAuMC4xOjE2NDQzCiAgY2EtY2VydDogIiIKICBuYW1lc3BhY2U6IGNvbnRyb2xsZXItazEKICByZW1vdGUtcG9ydDogIjE3MDcwIgogIHNlcnZpY2U6IGNvbnRyb2xsZXItc2VydmljZQogIHNlcnZpY2UtYWNjb3VudC10b2tlbjogZXlKaGJHY2lPaUpTVXpJMU5pSXNJbXRwWkNJNklteE1lV1ZJVFd3Mk56WjNTbmhUTTNwUlVuWmFRMUl3VDJOM1JFaEhUMkZtTmtreFJHZFhXbVV6UjNjaWZRLmV5SnBjM01pT2lKcmRXSmxjbTVsZEdWekwzTmxjblpwWTJWaFkyTnZkVzUwSWl3aWEzVmlaWEp1WlhSbGN5NXBieTl6WlhKMmFXTmxZV05qYjNWdWRDOXVZVzFsYzNCaFkyVWlPaUpqYjI1MGNtOXNiR1Z5TFdzeElpd2lhM1ZpWlhKdVpYUmxjeTVwYnk5elpYSjJhV05sWVdOamIzVnVkQzl6WldOeVpYUXVibUZ0WlNJNkltTnZiblJ5YjJ4c1pYSXRjSEp2ZUhraUxDSnJkV0psY201bGRHVnpMbWx2TDNObGNuWnBZMlZoWTJOdmRXNTBMM05sY25acFkyVXRZV05qYjNWdWRDNXVZVzFsSWpvaVkyOXVkSEp2Ykd4bGNpMXdjbTk0ZVNJc0ltdDFZbVZ5Ym1WMFpYTXVhVzh2YzJWeWRtbGpaV0ZqWTI5MWJuUXZjMlZ5ZG1salpTMWhZMk52ZFc1MExuVnBaQ0k2SWpObE5HWTNZek5qTFRsaVpUVXROR1V4TnkxaVl6RXpMV001TnpaaFpESXpZV1l3WVNJc0luTjFZaUk2SW5ONWMzUmxiVHB6WlhKMmFXTmxZV05qYjNWdWREcGpiMjUwY205c2JHVnlMV3N4T21OdmJuUnliMnhzWlhJdGNISnZlSGtpZlEuUTFoQThNUTh3TDQxcEtUYlZwYk43dFZZN0pyX2UzQ0pYa1Q4aVZzdERtTG1DMTE4cndlZzNuQVlJX282ZmdJb1lOODFYcDFWdUlFUjdBWEo2RGQxMmo2WFhuWEw0R3ZVRkQ3bjdwWWtQdE9ZZmhKSmpwdENINFJaWW1HQ1FobUdsVGJTZzdkUF9fVDFBN194NWNKbm83V0RHM3pDSUlJMllIdVl5ZkxrY2lHaXlzMWh2QllOQS04Z3h5anZ4RU12NUV6OFNNaEJkQ3dOaVY1em5nVjJVblA4R1F3WExHYjBkUzAyZlFzZTBCZHpfWHFBcjhSa3d4dlh1a2FtX3dHRWoyX2tBVllaN3VEbUVzLXA1VU15RjBXZDJObEFyUzJyWDVaT3czVU5pak5HN1p0VV9hUl81UF8yWkpXRG05MEJVS2dEalRvQ3FUZlNqYzdMeHpjY1BnCgAA
Enter a new password:
Confirm password:
Enter a name for this controller [k1]:
Initial password successfully set for admin1.

Welcome, admin1. You are now logged into "k1".

There are no models available. You can add models with
"juju add-model", or you can ask an administrator or owner
of a model to grant access to that model with "juju grant".

$ cat $JUJU_DATA/controllers.yaml | yq .controllers.k1."proxy-config"
type: kubernetes-port-forward
config:
  api-host: https://10.152.183.1:443
  ca-cert: |
    -----BEGIN CERTIFICATE-----
    MIIDDzCCAfegAwIBAgIUJMOPln3PfMts2ouNdSGXBiD2fzswDQYJKoZIhvcNAQEL
    ...
    HuWn0NUjyWvAmDs2CRJo+GkNbA==
    -----END CERTIFICATE-----
  namespace: controller-k1
  remote-port: "17070"
  service: controller-service
  service-account-token: eyJhbGciOiJSUzI1NiIsImtpZCI6Imx...TfSjc7LxzccPg
```

## Documentation changes

No

## Bug reference

No
